packages/compiler/src/server/compile-service.ts

```diff
@@ -174,12 +177,50 @@ export function createCompileService({
       if (serverHost.throwInternalErrors) {
         throw err;
       }
+
+      let uri = document.uri;
+      let range = Range.create(0, 0, 0, 0);
+      if (err.info.kind === "emitter" && configFilePath) {
+        const sourceFile = await serverHost.compilerHost.readFile(configFilePath);
+        const emitterName = err.info.metadata.name;
+        const yamlDoc = parseDocument(sourceFile.text, {
+          keepSourceTokens: true,
+        });
+        if (yamlDoc) {
+          let emitOffset = 0;
+          const emitNode = yamlDoc.get("emit");
+          if (isSeq(emitNode) && emitNode.srcToken?.type === "block-seq") {
+            const found = emitNode.srcToken.items.find(
+              (item) =>
+                item.value &&
+                "source" in item.value &&
+                item.value.source.replace(/^['"]|['"]$/g, "") === emitterName,
+            )?.value;
+            if (found) {
+              emitOffset = found.offset;
+            }
+          }
+
+          if (emitOffset > 0) {
+            const lineAndChar = sourceFile.getLineAndCharacterOfPosition(emitOffset + 1);
+
+            uri = fileService.getURL(configFilePath);
+            range = Range.create(
+              lineAndChar.line,
+              lineAndChar.character,
+              lineAndChar.line,
+              lineAndChar.character + emitterName.length,
+            );
+          }
+        }
+      }
+
       serverHost.sendDiagnostics({
-        uri: document.uri,
+        uri,
         diagnostics: [
           {
             severity: DiagnosticSeverity.Error,
-            range: Range.create(0, 0, 0, 0),
+            range,
```

Comment:
So, this actually maybe doesn't feel like the right thing here.

Why would the error happen in the emitter? how do you know that? it could be a compiler error (specially as the message shows that below)

Reply:
you should be able to do it if you check if the error is `ExternalError` and has an `info` on it with different possible sources.